service/controller/v24/resource/peerrolearn/create.go

```diff
@@ -0,0 +1,33 @@
+package peerrolearn
+
+import (
+	"context"
+
+	"github.com/giantswarm/microerror"
+
+	"github.com/giantswarm/aws-operator/service/controller/v24/key"
+)
+
+func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	cr, err := key.ToCustomObject(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	{
+		r.logger.LogCtx(ctx, "level", "debug", "message", "finding control plane peer role arn")
+
+		err = r.addPeerRoleARNToContext(ctx, cr)
+		if IsNotFound(err) {
+			r.logger.LogCtx(ctx, "level", "debug", "message", "did not find control plane peer role arn")
+			r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
```

Comment:
Should the resource be canceled here?  Or is it being canceled because nil is returned?

Reply:
> Default resources implement `EnsureCreated` and `EnsureDeleted`. In order to
> stop processing here it is good enough to simply `return` within the code. The
> resources being configured after the canceled resource are then executed and the
> canceled resource is executed again on the next reconciliation loop once the
> resync period kicks in again.

See https://github.com/giantswarm/operatorkit/blob/master/docs/control_flow_primitives.md#default-resources.

Reply:
Ack. Its the simple resource so that's all that's needed.

Sorry about the delayed reply. I've got a long lag getting mails from github for some reason.

Reply:
All good. We got that. <3